src/Analyzers/VisualBasic/Tests/AddRequiredParentheses/AddRequiredParenthesesTests.vb

```diff
@@ -8,162 +8,204 @@ Imports Microsoft.CodeAnalysis.Diagnostics
+        Private Const RequireAllParenthesesForClarity As String = "[*]
+dotnet_style_parentheses_in_arithmetic_binary_operators = always_for_clarity
+dotnet_style_parentheses_in_relational_binary_operators = always_for_clarity
+dotnet_style_parentheses_in_other_binary_operators = always_for_clarity
+dotnet_style_parentheses_in_other_operators = never_if_unnecessary
+"
+        Private Const RequireOtherBinaryParenthesesForClarity As String = "[*]
+dotnet_style_parentheses_in_arithmetic_binary_operators = never_if_unnecessary
+dotnet_style_parentheses_in_relational_binary_operators = never_if_unnecessary
+dotnet_style_parentheses_in_other_binary_operators = always_for_clarity
+dotnet_style_parentheses_in_other_operators = never_if_unnecessary
+"
+        Private Const RequireArithmeticBinaryParenthesesForClarity As String = "[*]
+dotnet_style_parentheses_in_arithmetic_binary_operators = always_for_clarity
+dotnet_style_parentheses_in_relational_binary_operators = never_if_unnecessary
+dotnet_style_parentheses_in_other_binary_operators = never_if_unnecessary
+dotnet_style_parentheses_in_other_operators = never_if_unnecessary
+"
+        Private Shared Async Function VerifyCodeFixAsync(code As String, fixedCode As String, editorConfig As String) As Task
+            Await New VerifyVB.Test With
+            {
+                .TestCode = code,
+                .FixedCode = fixedCode,
+                .EditorConfig = editorConfig
+            }.RunAsync()
+        End Function
 
-        Friend Overrides Function CreateDiagnosticProviderAndFixer(Workspace As Workspace) As (DiagnosticAnalyzer, CodeFixProvider)
-            Return (New VisualBasicAddRequiredParenthesesForBinaryLikeExpressionDiagnosticAnalyzer(), New AddRequiredParenthesesCodeFixProvider())
+        Private Shared Async Function VerifyNoCodeFixAsync(code As String, editorConfig As String) As Task
+            Await New VerifyVB.Test With
+            {
+                .TestCode = code,
+                .FixedCode = code,
+                .EditorConfig = editorConfig
+            }.RunAsync()
         End Function
 
         <Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddRequiredParentheses)>
         Public Async Function TestArithmeticPrecedence() As Task
-            Await TestInRegularAndScript1Async(
+            Await VerifyCodeFixAsync(
 "class C
     sub M()
-        dim x = 1 + 2 $$* 3
+        dim x = 1 + 2 [|*|] 3
     end sub
 end class",
 "class C
     sub M()
         dim x = 1 + (2 * 3)
     end sub
-end class", parameters:=New TestParameters(options:=RequireAllParenthesesForClarity))
+end class", RequireAllParenthesesForClarity)
         End Function
 
         <Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddRequiredParentheses)>
         Public Async Function TestNoArithmeticOnLowerPrecedence() As Task
-            Await TestMissingAsync(
+            Await VerifyCodeFixAsync(
```

Comment:
Shouldn't this also be invoking `VerifyNoCodeFixAsync` like the next test?

Reply:
@mavasani The old testing library was invoking the codefix exactly at `+` operator, and there was no diagnostic.

The new testing library checks diagnostics in the whole provided tree, so it found a diagnostic that has a fix.